### PR TITLE
doc: document Mediator's custom behaviors

### PR DESCRIPTION
**Pull request by FabioPinheiro**
https://github.com/input-output-hk/atala-prism-mediator/blob/doc/Error_Handling_Table/Mediator-Error_Handling.md

### CODE REVIEW

**Comment by FabioPinheiro** [comment]
https://github.com/input-output-hk/atala-prism-mediator/blob/doc/Error_Handling_Table/Mediator-Error_Handling.md

File: Mediator-Error_Handling.md [hunk 1]
```diff
@@ -1,63 +1,121 @@
...
+
+
+### Considerations
+
+In the Atala PRISM Mediator when a problem happens we try to:
+- logs the error
+- send a problem report according to the table above
+- store the problem report and error info on the BD. 
+
+All three points SHOULD be traceable to the initial call (by a call ID) and by the MsgID (hash SHA251 of the encrypted message).
```
Comment (mineme0110) [inline Mediator-Error_Handling.md]: SHOULD is capital case?
call ID ? is this  XRequestID ?

**Review by mineme0110 — APPROVED** [review]

File: Mediator-Error_Handling.md [hunk 2]
```diff
@@ -1,63 +1,121 @@
 # Error Handling
 
-https://identity.foundation/didcomm-messaging/spec/#problem-reports
 
-On each step of all our protocols processing, when something wrong is happening, we need to:
-Goals
+On each step of all our protocols processing, when something wrong is happening, we try to:
```
Comment (petevielhaber) [inline Mediator-Error_Handling.md:4]: On each step of our protocol processing, when something wrong is happening, we try to:

File: Mediator-Error_Handling.md [hunk 3]
```diff
@@ -1,63 +1,121 @@
...
+    - M5A - If the connection support live delivery respond with a [`status`](https://didcomm.org/messagepickup/3.0/status) message.
+    - M5B - If the connection is incapable of live delivery send a problem report `e.m.live-mode-not-supported`
+
+
+
+### Considerations
+
+In the Atala PRISM Mediator, when an issue arises, we undertake the following steps:
+* Log the error.
+* Dispatch a problem report as outlined in the aforementioned table.
```
Comment (petevielhaber) [inline Mediator-Error_Handling.md:108]: * Dispatch a problem report as outlined in the table above.

File: Mediator-Error_Handling.md [hunk 4]
```diff
@@ -1,63 +1,121 @@
...
+* Record the problem report and error information in the database.
+Every one of these steps should be traceable back to the initial call via a XRequestID,
```
Comment (petevielhaber) [inline Mediator-Error_Handling.md:110]: Every one of these steps should be traceable back to the initial call via an XRequestID, and by the MsgID (the SHA-251 hash of the encrypted message).

**Review by petevielhaber — COMMENTED** [review]
Suggested changes

**Pull request merged**